radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyLedgerEntryStore.java

```diff
@@ -265,6 +271,20 @@ public ByteBuffer verifyVirtualSubstate(SubstateId substateId)
 		return BerkeleyLedgerEntryStore.this.openIndexedCursor(null, index);
 	}
 
+	@Override
+	public Optional<RawSubstateBytes> get(SystemMapKey mapKey) {
+		var key = new DatabaseEntry(mapKey.array());
+		var substateId = new DatabaseEntry();
+		var result = mapDatabase.get(null, key, substateId, null);
+		if (result != SUCCESS) {
+			return Optional.empty();
+		}
+
+		var substate = loadSubstate(null, SubstateId.fromBytes(substateId.getData())).orElseThrow();
+		var substateBytes = new RawSubstateBytes(substateId.getData(), substate.array());
+		return Optional.of(substateBytes);
```

Comment:
Combination of `.orElseThrow()` with return value wrapped into `Optional` looks inconsistent. Perhaps method return value should be `Result`?

Reply:
It's not inconsistent in that `.orElseThrow` indicates that an unexpected error occurred which can occur if storage was changed from underneath program for example. This needs to be explicitly different from an empty or present Optional.